tests/job.go

```diff
@@ -17,30 +17,54 @@ import (
 func WaitForJobToSucceed(virtClient *kubecli.KubevirtClient, job *batchv1.Job, timeoutSec time.Duration) {
 	EventuallyWithOffset(1, func() bool {
 		job, err := (*virtClient).BatchV1().Jobs(job.Namespace).Get(job.Name, metav1.GetOptions{})
-		ExpectWithOffset(1, err).ToNot(HaveOccurred())
-		Expect(job.Status.Failed).NotTo(Equal(*job.Spec.BackoffLimit), "Job was expected to succeed but failed")
-		return job.Status.Succeeded > 0
+		ExpectWithOffset(2, err).ToNot(HaveOccurred())
+		for _, c := range job.Status.Conditions {
+			switch c.Type {
+			case batchv1.JobComplete:
+				if c.Status == k8sv1.ConditionTrue {
+					return true
+				}
+			case batchv1.JobFailed:
+				ExpectWithOffset(2, c.Status).To(Equal(k8sv1.ConditionTrue), "Job should succeed")
+			}
+		}
+		return false
 	}, timeoutSec*time.Second, 1*time.Second).Should(BeTrue(), "Job should succeed")
 
 }
 
 func WaitForJobToFail(virtClient *kubecli.KubevirtClient, job *batchv1.Job, timeoutSec time.Duration) {
-	EventuallyWithOffset(1, func() int32 {
+	EventuallyWithOffset(1, func() bool {
 		job, err := (*virtClient).BatchV1().Jobs(job.Namespace).Get(job.Name, metav1.GetOptions{})
-		ExpectWithOffset(1, err).ToNot(HaveOccurred())
-		Expect(job.Status.Succeeded).ToNot(BeNumerically(">", 0), "Job should not succeed")
-		return job.Status.Failed
-	}, timeoutSec*time.Second, 1*time.Second).Should(BeNumerically(">", 0), "Job should fail")
+		ExpectWithOffset(2, err).ToNot(HaveOccurred())
+		for _, c := range job.Status.Conditions {
+			switch c.Type {
+			case batchv1.JobFailed:
+				if c.Status == k8sv1.ConditionTrue {
+					return true
+				}
+			case batchv1.JobComplete:
+				ExpectWithOffset(2, c.Status).To(Equal(k8sv1.ConditionTrue), "Job should fail")
+			}
+		}
+		return false
+	}, timeoutSec*time.Second, 1*time.Second).Should(BeTrue(), "Job should fail")
 }
 
-func RenderJob(name string, cmd, args []string) *batchv1.Job {
+// NewJob creates a job configuration that runs a single Pod.
+// A name is used for the job & pod while the command and its arguments are passed to the pod for execution.
+// In addition, the following arguments control the job behavior:
+// retry: The number of times the job should try and run the pod.
+// ttlAfterFinished: The period of time between the job finishing and its auto-deletion.
+// timeout: The overall time at which the job is deleted, regardless of it finishing or not.
+func NewJob(name string, cmd, args []string, retry, ttlAfterFinished int32, timeout int64) *batchv1.Job {
```

Comment:
1. You can already achieve this by modifying `job` which is returned (having reasonable defaults and be open to modification seems reasonable to me). 
2. Maybe we can take `options` as args if you feel this is needed. Then we can have something like `NewJob("netcat", []string{"/bin/bash", "-c"}, check, withRetry(3))`

Reply:
The job arguments that control its behavior are limited to 3 at the moment. I do not see others used and it seems simpler to explicitly require them.
As I need to use this in a following PR, with different values, I do not see the added value in exposing implicit defaults or complicating this with "with*" options.

If we'll need to edit the jobs object after it is returned, we will loose the helper purpose of not requiring the caller to know how a job looks like deep in (i.e. abstraction).

Reply:
Can you add a note that ttlAfterFinished needs to be large enough to collect logs and state if the test fails? Maybe we can set some reasonable default on this one.

Reply:
Sure, I will add the note.
Regarding the default, that could work, the downside is that if someone will want to customize it in the future, we will have to either change the signature or create a new function. It is hard to predict the future.

If it is very important to you, I will use a default instead of the argument for the ttl, otherwise I will just create constants to specify recommended defaults.

Reply:
Both works for me. I just want to make sure that the user will do the right thing and we will have the debug info.

Reply:
Done